.changes/unreleased/ENHANCEMENTS-20231128-143901.yaml

```diff
@@ -0,0 +1,6 @@
+kind: ENHANCEMENTS
+body: 'helper/resource: Automatically add `required_providers` configuration to `TestStep.Config`
+  configuration when using Terraform 1.6.* and above'
```

Comment:
Good consideration callout in the PR description about JSON, since we currently treat JSON as "you know what you're doing" and our future functionality testing documentation will always show Terraform configuration language, I think just putting a quick "Terraform language" before the word "configuration" here will suffice. Also if we do update the check to 1.0, that will need to be updated here as well. 😄